storm-core/test/jvm/org/apache/storm/security/auth/AuthUtilsTest.java

```diff
@@ -0,0 +1,227 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.security.auth;
+
+import java.io.IOException;
+import java.io.File;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import javax.security.auth.Subject;
+
+import org.apache.commons.codec.binary.Hex;
+import org.apache.storm.Config;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class AuthUtilsTest {
+
+    @Test(expected=IOException.class)
+    public void getOptionsThrowsOnMissingSectionTest() throws IOException {
+        Configuration mockConfig = Mockito.mock(Configuration.class);
+        AuthUtils.get(mockConfig, "bogus-section", "");
+    }
+
+    @Test
+    public void getNonExistentSectionTest() throws IOException {
+        Map<String, String> optionMap = new HashMap<String, String>();
+        AppConfigurationEntry entry = Mockito.mock(AppConfigurationEntry.class);
+        
+        Mockito.<Map<String, ?>>when(entry.getOptions()).thenReturn(optionMap);
+        String section = "bogus-section";
+        Configuration mockConfig = Mockito.mock(Configuration.class);
+        Mockito.when(mockConfig.getAppConfigurationEntry(section))
+               .thenReturn(new AppConfigurationEntry[] {entry});
+        Assert.assertNull(
+                AuthUtils.get(mockConfig, section, "nonexistent-key"));
+    }
+
+    @Test
+    public void getFirstValueForValidKeyTest() throws IOException {
+        Map<String, String> optionMap = new HashMap<String, String>();
+        optionMap.put("existent-key", "foo");
+        AppConfigurationEntry entry = Mockito.mock(AppConfigurationEntry.class);
+        
+        Mockito.<Map<String, ?>>when(entry.getOptions()).thenReturn(optionMap);
+        String section = "bogus-section";
+        Configuration mockConfig = Mockito.mock(Configuration.class);
+        Mockito.when(mockConfig.getAppConfigurationEntry(section))
+               .thenReturn(new AppConfigurationEntry[] {entry});
+        Assert.assertEquals(
+                AuthUtils.get(mockConfig, section, "existent-key"), "foo");
+    }
```

Comment:
The test logic is different with clojure's test-returns-first-value-for-valid-key